### ISSUE TITLE
Dashboard overview by build id

### ISSUE THREAD

**Issue by mathpaquette** [sorry-cypress/sorry-cypress#362]
- [X] I have searched issues for similar feature requests

## What is this feature about (1 sentence)
As a monorepo developer,
I want to have an overview of all affected projects by looking for a specific build id.

## Why is it needed? What is the value? For whom do we build it?
Especially when you deal with a monorepo, you have many applications getting tested in the same CI run,
you need to know if everything went smoothly or not before merging. By supporting that, it gives a much quicker overview instead of having to look into each project individually.

## High-level feature overview
Basically just a link like `/builds/<build id>/` listing all projects, runs and statuses using the build id. We can then drilldown more from there.

**Comment by mathpaquette** [comment]
@agoldis do you agree on this one? any guidance ?

**Comment by agoldis** [comment]
Let's make sure I got it right. You have a monorepo, each package in the monorepo users a different project id. CI assigns the same `ciBuildId` for each package, so you have the same `ciBuildId` across multiple projects.
```
my-app/
├─ packages/
│  ├─ app_a/
│  │  ├─ cypress.json (projectId: 101)
│  ├─ app_b/
│  │  ├─ cypress.json (projectId: 102)
│  ├─ app_c/
│  │  ├─ cypress.json (projectId: 103)

```

You want to have an aggregated cross-project view for runs with the same `ciBuildId`. If that's the case:
 
- it'd be a bit hard to discover - how do one gets to that page?
- shouldn't be too hard to implement - query would be `get all runs with cilBuildId`, I think it is kind of implemented already via `filters` in gql query 

---

As an alternative, please consider using a single project id and assign a different group for each package in the monorepo. 

That would work nicely with CI / hooks reporting and should work out of the box. E.g. `cy2 run --parallel --record --key xxx --group packageA --ciBuildId yyy`. A small adjustment to how specs are listed visually would make it more clear to review. I have implemented the following layout for https://currents.dev. 

<img width="1132" alt="Screen Shot 2021-05-28 at 12 18 04 PM" src="https://user-images.githubusercontent.com/1637928/120032337-e610af80-bfae-11eb-86f3-ca5132ac9b01.png">

**Comment by jdileonardo** [comment]
> it'd be a bit hard to discover - how do one gets to that page?

Tabulating the top nav to have an "All Projects" and an "All Builds" could be sufficient?

I'm in the same boat as well. I leverage a mono-repo structure using NX and based on a dep graph build certain projects ; navigating sorry-cypress based on CI-Build-ID vs individual project is more natural for my setup.

re: your suggestion, Group var appears to only be able to be set on the cmd line, which doesn't lend itself exactly to some of my CI setup. Dividing by project still gives individual module owner within the monorepo granularity to configure incoming/outcoming hooks per project.

my 2 cents, but i'd find this useful tbh.

**Comment by agoldis** [comment]
> Tabulating the top nav to have an "All Projects" and an "All Builds" could be sufficient?

Yeah, good idea!

Ok, if you find it useful, I am not one to stop you :) 

--- 

@jdileonardo 

> a mono-repo structure using NX

Is it working well for you? I am struggling to maintain a good devex for sorry-cypress project. In particular
- build dependencies
- dev / watch with dependenies
- TS and VSCode navigation

Currently looking at Rush and NX, but wanted to hear what's your experience so far?

**Comment by mathpaquette** [comment]
> > Tabulating the top nav to have an "All Projects" and an "All Builds" could be sufficient?
> 
> Yeah, good idea!
> 
> Ok, if you find it useful, I am not one to stop you :)
> 
> @jdileonardo
> 
> > a mono-repo structure using NX
> 
> Is it working well for you? I am struggling to maintain a good devex for sorry-cypress project. In particular
> 
> * build dependencies
> * dev / watch with dependenies
> * TS and VSCode navigation
> 
> Currently looking at Rush and NX, but wanted to hear what's your experience so far?

@agoldis I'm also using Nx. pretty darn good TBH. It uses cache and dependency tree to detect affected apps/libs. I like that. I got about 30 different apps within the same repo that's why the Builds view is very important for me as well.

**Comment by mathpaquette** [comment]
@jdileonardo @agoldis did you start working on this ? if not ill try something.

**Comment by agoldis** [comment]
@mathpaquette not yet

**Comment by maxfriedmann** [comment]
@mathpaquette can you share how you are using sorry-cypress within an nx workspace? I created https://github.com/sorry-cypress/sorry-cypress/issues/484 and am very interested in your solution.

**Comment by mathpaquette** [comment]
@maxfriedmann sorry for the late reply... I just realized this new message... Anyways.
The best way I found was to create for each e2e projects a new configuration entry for sorry-cypress. In our case, we support 2 different suites (continuous integration and post-deploy) and they have their own files so we create 2 distinct entries but it doesn't matter much how you implement it.

Quick config example:

```
my-app-e2e.targets.e2e.configurations."your-suite": {
ciBuildId: ""
...
} 
``

Then you just need to call in CI something like: nx affected:e2e --configuration your-suite

Hope it helps.

**Comment by mathpaquette** [comment]
can we reopen this one

**Comment by mathpaquette** [comment]
@agoldis

**Comment by agoldis** [comment]
@mathpaquette  It's been open for 1.5+ years, why do you want it to be open

**Comment by mathpaquette** [comment]
well, I need a ticket to commit on haha

**Comment by mathpaquette** [comment]
mostly just to be sure we still want the feature to be merged.

**Comment by mathpaquette** [comment]
@agoldis anyway, I'm on it.. i'll have a PR soon. The issue has been created 1.5 year ago and my daughter is 1.5 too. I think there's a correlation between both ;-) . Have a nice one.

**Comment by agoldis** [comment]
Haha, congrats! My daughter is 2.5 btw

**Comment by luengnat** [comment]
It should also be able to group by some manual tagging not just by project?

**Comment by mathpaquette** [comment]
> It should also be able to group by some manual tagging not just by project?

Umm. I don't think so. Right now, we are grouping runs by `meta.ciBuildId` (which should be a unique value per CI build) and each runs have an associated project.

The goal of this feature is to give you an overview of each CI builds at once without having to open each projects one by one.
https://github.com/sorry-cypress/sorry-cypress/pull/639#issuecomment-1233568505
